I don't think this works.

I installed the modified version in PowerShell as follows:

    pipx install --suffix test visidata@git+https://github.com/Kreijstal/visidata.git@patch-1

`--suffix test` appends `test` to the executable names to avoid clobbering your existing Visidata installation.
The installation spec uses Pip's [VCS support](https://pip.pypa.io/en/latest/topics/vcs-support/#git) to specify the required git source and branch name.

I confirm that it does not install `windows-curses` as follows:

```
❯ pipx runpip visidatatest list
Package         Version
--------------- -----------
pip             25.1.1
python-dateutil 2.9.0.post0
six             1.17.0
visidata        3.2.dev0
```

When I run it in PowerShell, I get the following error:

```
❯ vdtest
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "c:\users\dwales\.local\bin\vdtest.exe\__main__.py", line 4, in <module>
  File "C:\Users\dwales\pipx\venvs\visidatatest\Lib\site-packages\visidata\__init__.py", line 32, in <module>
    from .vdobj import *
  File "C:\Users\dwales\pipx\venvs\visidatatest\Lib\site-packages\visidata\vdobj.py", line 2, in <module>
    import curses
  File "C:\Users\dwales\AppData\Local\Programs\Python\Python312\Lib\curses\__init__.py", line 13, in <module>
    from _curses import *
ModuleNotFoundError: No module named '_curses'
```

As you can see, I'm using Python 3.12 here. I get the same error if I run `import curses` directly in a REPL.

So far as I can tell, the Windows build of Python (at least for 3.12 and below) does not include the curses library, so even if Windows ConPTY now theoretically enables curses to work on Windows, it's not yet bundled with Python.

I expect we need to wait for this CPython issue to be resolved:
https://github.com/python/cpython/issues/85796

--- a/setup.py
+++ b/setup.py
@@ -2,11 +2,20 @@
 
 from setuptools import setup
 import platform
+import sysconfig
+
 
 
 # tox can't actually run python3 setup.py: https://github.com/tox-dev/tox/issues/96
 # from visidata import __version__
 __version__ = "3.2dev"
+install_requires = [
+    "python-dateutil",
+    'importlib_resources; python_version<"3.9"',
+]
+
+if not sysconfig.get_platform().startswith("mingw"):  # 2757
+    install_requires += ['windows-curses >= 2.4.1; platform_system == "Windows"']   # 2119
 
 setup(
     name="visidata",
@@ -25,10 +34,7 @@ setup(
                             "visidata=visidata.main:vd_cli"],
     },
     py_modules=["visidata"],
-    install_requires=[
-        "python-dateutil",
-        'importlib_resources; python_version<"3.9"',
-    ],
+    install_requires=install_requires,
     packages=[
         "visidata",
         "visidata.loaders",

